kf_lib_data_ingest/etl/ingest_pipeline.py

```diff
@@ -11,6 +11,10 @@
 from kf_lib_data_ingest.common.concept_schema import UNIQUE_ID_ATTR
```

Comment:
Not sure where to put this but it applies to this file. 

I think ingest should not fail if it is not possible to connect to the warehouse (due to missing URL, incorrect URL, offline db, etc). Ingest should continue as if it were not going to load into the warehouse. As long as the user is alerted in the log whether warehouse loading succeeded or not 

Consider something like this:

```python
    def run(self):
        """
        Entry point for data ingestion. Run ingestion in the top level
        exception handler so that exceptions are logged.
        """
        self.logger.info("BEGIN data ingestion.")
        self.stages = {}
        self._init_warehouse = False
        warehouse_success = False
        all_passed = True
        all_messages = []

        # Top level exception handler
        # Catch exception, log it to file and console, and exit
        try:
            # Iterate over all stages in the ingest pipeline
            output = None
            for stage in self._iterate_stages():
                # No more stages left in list of user specified stages
                if not self.stages_to_run:
                    break

                self.stages[stage.stage_type] = stage

                # Execute/run stage
                if stage.stage_type in self.stages_to_run:
                    self.stages_to_run.remove(stage.stage_type)

                    output = stage.run(output)

                    # Standard stage output validation
                    if stage.concept_discovery_dict:
                        passed, messages = self.check_stage_counts(stage)
                        all_passed = passed and all_passed
                        all_messages.extend(messages)

                # Load cached output and concept counts
                else:
                    self.logger.info(
                        "Loading previously cached output and concept counts "
                        f"from {type(stage).__name__}"
                    )
                    output = stage.read_output()
                    stage.read_concept_counts()

                # Push Extract/Transform stage outputs to warehouse
                warehouse_success = self._load_warehouse(stage, output)

            self._log_count_results(all_passed, all_messages)

            # Run user defined data validation tests
            all_passed = self.user_defined_tests() and all_passed

        except Exception as e:
            self.logger.exception(str(e))
            self.logger.info("Exiting.")
            sys.exit(1)

        # Log the end of the run
        if warehouse_success:
            self.logger.info('✅ Push to warehouse succeeded')
        else:
            self.logger.warning(
                '⚠️ Push to warehouse failed! See ingest log for details'
            )

        self.logger.info("END data ingestion")
        return all_passed

    def _load_warehouse(self, stage, output):
        """
        Push Extract/Transform stage outputs to warehouse
        """
        success = True
        try:
            if self.warehouse_db_url and isinstance(
                stage, (TransformStage, ExtractStage)
            ):
                # Initialize database if needed
                if self._init_warehouse:
                    init_study_db(
                        self.warehouse_db_url,
                        self.data_ingest_config.study
                    )
                    self._init_warehouse = True

                for df_name, df in output.items():
                    persist_df_to_study_db(
                        self.warehouse_db_url,
                        self.data_ingest_config.study,
                        type(stage).__name__,
                        df,
                        df_name,
                    )
                    self.logger.info(
                        f'Loaded {type(stage).__name__}: {df_name} '
                        'in warehouse'
                    )
        except Exception as e:
            self.logger.warning(
                f'⚠️ Push to warehouse failed! Caused by: {str(e)}'
            )
            success = False

        return success
```